File: src/PromptResponse.Core/Models/DocumentType.cs
namespace PromptResponse.Core.Models;

/// <summary>
/// Specifies the type of APR document.
/// </summary>
public enum DocumentType
{
    /// <summary>
    /// A blank template to be filled out.
    /// </summary>
    /// <remarks>
    /// Templates define the structure and prompts but typically have empty responses.
    /// When a user opens a template, they can choose to edit the template structure
    /// or fill it out as a form.
    /// </remarks>
    Template = 0,

    /// <summary>
    /// A form that has been filled out with responses.
    /// </summary>
    /// <remarks>
    /// Filled forms are based on templates and contain user-provided responses.
    /// When opening a filled form, the application goes directly to editing mode.
    /// </remarks>
    FilledForm = 1
}

File: java/src/main/java/io/promptresponse/models/DocumentType.java
package io.promptresponse.models;

import com.fasterxml.jackson.annotation.JsonValue;

public enum DocumentType {
    TEMPLATE("template"),
    FILLED_FORM("filledForm");

    private final String value;

    DocumentType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }

    public static DocumentType fromValue(String value) {
        for (DocumentType type : values()) {
            if (type.value.equals(value)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown document type: " + value);
    }
}
